tests/test_system.py

```diff
@@ -2,535 +2,64 @@
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_security():
-    module_name = "system.security"
-
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_serial():
-    module_name = "system.serial"
-
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_sfc():
-    module_name = "system.sfc"
-
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_tag():
-    module_name = "system.tag"
-
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_twilio():
-    module_name = "system.twilio"
-
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_user():
-    module_name = "system.user"
-
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_util():
-    module_name = "system.util"
-
-    try:
-        imported_module = importlib.import_module(module_name)
-    except ImportError:
-        assert False, "Failed to import module: {0}".format(module_name)
-
-    assert imported_module is not None, "Module {0} is None after import.".format(
-        module_name
-    )
-
-    _cleanup()
-
-
-def test_vision():
-    module_name = "system.vision"
-
+@pytest.mark.parametrize("test_module", _SYSTEM_MODULES)
+def test_system(test_module):
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (testing):** Ensure cleanup is effective in the context of parameterized tests.

Given that `_cleanup` is called after each module import, it's crucial to verify that this cleanup process does not inadvertently affect subsequent tests, especially in a parameterized setup. Consider isolating each test module's environment more robustly or ensuring that the cleanup process is scoped only to the changes made by the test.